Add support to Akida 1500

kmaincent opened the pull request:
These changes add support to Akida 1500 board.
Support only mmap operation for now.
The two BAR memories accessible are 0xFCC0_0000 to 0xFCFF_FFFF for BAR 2 and 0x2000_0000 to 0x203F_FFFF for BAR 4.

The mmap test has been updated accordingly and is still compatible with old Akida 1000 board.
Here is the test example:
  $ ./mmap_access /dev/akd1500_0 0xfcc00054 32
  0x10010114

Perform 32bit write, 0xcafedeca value at  0x20000020  (Memory):
  $ ./mmap_access /dev/akd1500_0 0x20000020 32 0xcafedeca
  $ ./mmap_access /dev/akd1500_0 0x20000020 32
  0xcafedeca

kmaincent committed:
akida: Handle kernel 5.17..5.19

Signed-off-by: Kory Maincent <kory.maincent@bootlin.com>

kmaincent committed:
akida: pcie: add support to Akida 1500

Add support to akida 1500 board. For now only mmap is supported.
It can map BAR_2 (0xFCC0_0000) and BAR_4 (0x2000_0000).
It uses a different mmap API than Akida 1000. The new API expect the
hardware BAR offset address as offset parameter. i.e. 0xFCC0_0000 to
0xFCFF_FFFF for BAR 2 and 0x2000_0000 to 0x203F_FFFF for BAR 4.

Signed-off-by: Kory Maincent <kory.maincent@bootlin.com>

kmaincent committed:
test: mmap: update mmap_access to new mmap API for Akida 1500

The new test file is directly using the offset pass from the user instead
of mapping the all PCI BAR. It mmap one page and seek or write the value
from that page. The old mmap API from Akida 1000 can still be used.

Signed-off-by: Kory Maincent <kory.maincent@bootlin.com>

kmaincent committed:
99-akida-pcie.rules: add akida 1500 rule

Signed-off-by: Kory Maincent <kory.maincent@bootlin.com>

dcorvoysier-brainchip commented:
I only went through the changes quickly, and noticed that:

- there are several changes not really related to the introduction of AKD1500 (change of wording, cases or indentation). I usually prefer having these in a separate commit,
- there is quite a lot of duplication in the driver code now: I think we could use a little refactoring to avoid that.

The pull-request has already been approved so I let @dirac3000 decide if this is ok to merge it as is.

kmaincent commented:
>     * there are several changes not really related to the introduction of AKD1500 (change of wording, cases or indentation). I usually prefer having these in a separate commit,

Are you talking about the renaming of Akida 1000 functions and defines? I can indeed have these in a separate commit.

>     * there is quite a lot of duplication in the driver code now: I think we could use a little refactoring to avoid that.

As there are several changes between the two boards, I think it is more readable to separate the functions or maybe we could add common subfunctions, it is as you prefer.

tengomucho commented:
I think splitting the renaming in a separate commit could help. If it is possible to have some common sub-function where there is some common code, it would be good too (I think for example in mmap functions).

kmaincent committed:
akida: rename few Akida 1000 functions

Rename few akida 1000 functions and define to prepare the support of
Akida 1500

Signed-off-by: Kory Maincent <kory.maincent@bootlin.com>

rchevalier-brainchip commented on akida-pcie-core.c:
```diff
@@ -894,6 +979,103 @@ static int akida_1000_probe(struct pci_dev *pdev, const struct pci_device_id *id
+
+	akida->pdev = pdev;
+
+	/* Disable ASPM L0s and L1 states as they cause device stop working */
```
Is that still true with AK1500? (just a question, no change required)
